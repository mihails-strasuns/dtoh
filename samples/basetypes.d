module basetypes;

extern(C) __gshared {
    byte a1;
    ubyte a2;
    short a3;
    ushort a4;
    int a5;
    uint a6;
    long a7;
    ulong a8;

    void* b1;
    int** b2;
    int[3] b3;

    float c1;
    double c2;
    real c3;

    cfloat d1;
    cdouble d2;
    creal d3;

    char e1;
    bool e2;
}
