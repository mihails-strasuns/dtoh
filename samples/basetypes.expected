#include <stdint.h>

// Used enum definitions:



// Struct forward declaration to resolve cycles:



// Function pointer types:



// Used struct definitions:



// Variable and function declarations:

int8_t a1;
uint8_t a2;
int16_t a3;
uint16_t a4;
int32_t a5;
uint32_t a6;
int64_t a7;
uint64_t a8;
void* b1;
int32_t** b2;
int32_t b3[3];
float c1;
double c2;
long double c3;
float _Complex d1;
double _Complex d2;
long double _Complex d3;
char e1;
_Bool e2;
